text/sigma-protocol/README.md

```diff
@@ -0,0 +1,76 @@
+Consider an example of prover proving knowledge of `x` in g<sup>`x`</sup>. Lets call g<sup>`x`</sup> as Y. Only the prover knows `x` but both prover and verifier know `Y` and `g`. 
+- prover picks a random number `r`, creates commitment T = g<sup>`r`</sup> and sends T to the verifier.
+- verifier sends challenge c to prover.
+- prover computes response `s = r - c*x` and sends `s` to verifier.
+- verifier checks if g<sup>`s`</sup>*Y<sup>`c`</sup> equals T. If it does then verifier is convinced of prover's knowledge of `x`.
+
+In the non-interactive version prover will hash g, Y and T to compute challenge c. There are more details and variations for the above protocol which can be seen [here](https://en.wikipedia.org/wiki/Proof_of_knowledge#Sigma_protocols).  Similar protocols can be developed more more complex relations that can be seen [here](https://medium.com/@loveshharchandani/zero-knowledge-proofs-with-sigma-protocols-91e94858a1fb).
+
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+This section will describe how such proofs of knowledge are to be constructed in reference to [PS signatures](https://eprint.iacr.org/2015/525), 6.2. The code for the proofs of knowledge are [here](https://github.com/lovesh/signature-schemes/blob/master/ps/src/pok.rs).  
+- For the 1st step (commitment), the prover first creates a `ProverCommitting` object. This object holds the generators and random values used during this step. 
+- Prover calls `ProverCommitting::commit` for each proof of knowledge that he wishes to do. This function takes a generator (`g`) and optionally takes a randomness. If not supplied, it creates its own.
+- Once the prover has committed for all its secrets, it calls `ProverCommitting::finish` method that results in creation of `ProverCommitted` object after consuming `ProverCommitting`. 
+- `ProverCommitted` marks the end of commitment phase and has the final commitment.
+- Now if this is a non-interactive protocol and the prover wishes to compute the challenge, `ProverCommitted` has a method `ProverCommitted::gen_challenge` to generate the challenge by hashing all generators and commitment. It is optional to use this method as the challenge may come from a super-protocol or from verifier. It takes a vector of bytes that it includes for hashing for computing the challenge. In an interactive proof, the prover can simply receive the challenge.
+- For the 3rd step, the prover calls `ProverCommitted::gen_proof` that takes the challenge (which was either computed by hashing or received from verifier) and the secrets (`x`) to compute a `Proof`. 
+- Now the verifier can verify the proof by calling `Proof::verify`. 
+
+To see the above protocol in action, have a look at this [test](https://github.com/lovesh/signature-schemes/blob/master/ps/src/pok.rs#L348).  
```

Comment:
Again, consider moving the code into the same folder as the README.md that is the RFC. This pattern has been followed with reference code for some of the RFCs in Aries.